packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -27,11 +27,25 @@ const CookieDefaults = {
   encode: val => encodeURIComponent(typeof val === 'string' ? val : JSON.stringify(val))
 } satisfies CookieOptions<any>
 
+let timeoutIDToClear: NodeJS.Timeout
```

Comment:
I didn't found another way to clear the timeout when **useCookie** is unmounted, because when you set the timeout inside customRef you can't have acces to his ID, I've tried to return the value of the cookie and the timeout from customRef but that was a bad idea.

I think creating a registry object is a good idea.
But in which file should I declare this variable ?